browser/ai_chat/ai_chat_tab_helper.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_AI_CHAT_AI_CHAT_TAB_HELPER_H_
+#define BRAVE_BROWSER_AI_CHAT_AI_CHAT_TAB_HELPER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/observer_list.h"
+#include "brave/browser/ai_chat/ai_chat.mojom.h"
+#include "brave/browser/ai_chat/ai_chat_api.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace ui {
+class AXTree;
+}  // namespace ui
+
+class AIChatTabHelper : public content::WebContentsObserver,
+                        public content::WebContentsUserData<AIChatTabHelper> {
+  using OnArticleSummaryCallback =
+      base::OnceCallback<void(const std::u16string& summary,
+                              bool is_from_cache)>;
+
```

Comment:
If this is intended to be private, let's move this to `private:` section below.

Reply:
moved